gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthorizationService.js

```diff
@@ -344,6 +353,46 @@ var fluid = fluid || require("infusion");
     // ==== End of addAuthorization()
 
     /**
+     * A common utility function shared by checkAuthDecision() and doAdd()
+     * This function retrieves and returns an authorization decision id that associates with the given gpii token and the client.
+     * If the authorization decision does not exist, this function will create one and return the created authorization decision id.
+     * @param dataStore {Object} An instance of gpii.oauth2.dbDataStore
+     * @param codeGenerator {Object} An instance of gpii.oauth2.codeGenerator
+     * @param clientId {String} A client id
+     * @param redirectUri {String} A redirect URI
+     * @param selectedPreferences {Object} A preference set
+     * @return {Promise} The resolved promise value is in the structure of:
+     * {
+     *     id: [authDecisionId]
```

Comment:
Do we mean that the authDecisionId is held in an array? Perhaps angle brackets might cause less confusion